digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/SpeedLimitService.scala

```diff
@@ -379,29 +363,20 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
       .map(isSeparableValidation)
       .get
 
-    withDynTransaction {
-      val idUpdated = updateSpeedLimitValue(id, valueTowardsDigitization, username, municipalityValidation).get
+    val (idUpdated, newId) = withDynTransaction {
+      val idUpdated = updateSpeedLimit(id, valueTowardsDigitization, username, None, None, municipalityValidation).get
       dao.updateSideCode(idUpdated, SideCode.TowardsDigitizing)
       val newId = dao.createSpeedLimit(username, speedLimit.linkId, Measures(speedLimit.startMeasure, speedLimit.endMeasure), SideCode.AgainstDigitizing, valueAgainstDigitization, None, linkSource = speedLimit.linkSource).get
 
-      Seq(loadSpeedLimit(idUpdated).get, loadSpeedLimit(newId).get)
+      (idUpdated, newId)
     }
-  }
-
-  /**
-    * Returns speed limits by municipality. Used by IntegrationApi speed_limits endpoint.
-    */
-  def get(municipality: Int): Seq[SpeedLimit] = {
-    val (roadLinks, changes) = roadLinkServiceImplementation.getRoadLinksWithComplementaryAndChangesFromVVH(municipality)
-    withDynTransaction {
-      getFilledTopologyAndRoadLinks(roadLinks, changes)._1
+      Seq(getSpeedLimitById(idUpdated).get, getSpeedLimitById(newId).get)
```

Comment:
We can replace this by getSpeedLimitByIds(Seq(idUpdated, newId)) and reduce the database access